src/bp/ui-studio/src/web/views/OneFlow/sidePanel/index.tsx

```diff
@@ -151,52 +172,89 @@ const SidePanelContent: FC<Props> = props => {
-              editQnA={editQnA}
-              topics={props.topics}
-              exportTopic={exportTopic}
-              fetchTopics={props.fetchTopics}
-            />
-          </SidePanelSection>
-
-          <SidePanelSection label={lang.tr('library')}>
-            <SearchBar
-              icon="filter"
-              placeholder={lang.tr('studio.flow.sidePanel.filterLibrary')}
-              onChange={setLibraryFilter}
-            />
-            <Library filter={libraryFilter} />
-          </SidePanelSection>
+          <Navbar className={style.topicsNavbar}>
+            <NavbarGroup>
+              <Tabs onChange={onTabChanged}>
+                <Tab id="topics" title={lang.tr('topics')} />
+                <Tab id="library" title={lang.tr('library')} />
+              </Tabs>
+            </NavbarGroup>
+            {props.permissions.includes('create') && (
+              <NavbarGroup align={Alignment.RIGHT}>
+                <Tooltip content={lang.tr('studio.flow.sidePanel.importTopic')}>
+                  <Button icon="import" onClick={() => setImportModalOpen(true)} />
+                </Tooltip>
+                <Tooltip content={lang.tr('studio.flow.sidePanel.addTopic')}>
+                  <Button icon="plus" onClick={() => createTopic()} />
+                </Tooltip>
+              </NavbarGroup>
+            )}
+          </Navbar>
+
+          {currentTab === 'topics' && (
+            <React.Fragment>
+              <SearchBar
+                icon="filter"
+                placeholder={lang.tr('studio.flow.sidePanel.filterTopicsAndWorkflows')}
+                onChange={setTopicFilter}
+              />
+              <TopicList
+                readOnly={props.readOnly}
+                canDelete={canDelete}
+                flows={props.flowsName}
+                qnaCountByTopic={props.qnaCountByTopic}
+                goToFlow={goToFlow}
+                deleteFlow={props.deleteFlow}
+                duplicateFlow={duplicateFlow}
+                currentFlow={props.currentFlow}
+                editWorkflow={editWorkflow}
+                createWorkflow={createWorkflow}
+                exportWorkflow={exportWorkflow}
+                importWorkflow={importWorkflow}
+                renameFlow={props.renameFlow}
+                updateFlow={props.updateFlow}
+                filter={topicFilter}
+                editTopic={editTopic}
+                editQnA={editQnA}
+                topics={props.topics}
+                exportTopic={exportTopic}
+                fetchTopics={props.fetchTopics}
+                fetchFlows={props.fetchFlows}
+                focusedText={focusedText}
+                expandedPaths={expandedPaths}
+                newPath={newPath}
+                setFocusedText={x => {
+                  setFocusedText(x)
+                  props.fetchTopics()
+                }}
+                setNewPath={setNewPath}
+                onExpandToggle={onExpandToggle}
+              />
```

Comment:
There's way too much properties on that component. Can you connect TopicList directly to the store, and access the properties which are passed directly ?

Reply:
I have to agree with @allardy on this. That property list is spectacular. This component has to be connected to the state. Most of the store functions are only bounded to be passed down why not bind them directly to the component